### DISCUSSION TITLE
PKCE on version 4

### DISCUSSION THREAD

**Discussion by Herber230** [nextauthjs/next-auth#3725] [Help] [answered]
I am trying to migrate from 3.27 to 4.1 with Okta as a provider.

I have question because I haven't found any PKCE related docs. In the previous version there was a field called `protection` for the provider declaration, so I can define PKCE as my authentication flow.

```
export default NextAuth({
  providers: [
    Providers.Okta({
      protection: 'pkce',
      idToken: true,
      clientId,
      authorizationUrl,
      accessTokenUrl,
      scope: 'openid email services',
    }),
  ],
});
```

My question is:  Why there isn't available this field in the new constructor? How can I define my PKCE auth flow now?

> import OktaProvider from 'next-auth/providers/okta';

**Comment by balazsorban44** [comment, marked as answer]
it just got renamed to `checks`

https://github.com/nextauthjs/next-auth/pull/2255

**Reply by Herber230** [reply]
Thanks
